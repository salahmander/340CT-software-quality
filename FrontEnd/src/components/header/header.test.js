import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom'

import Header from './header';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should render correctly in "debug" mode and match the snapshot', () => {
  const component = shallow(<Header debug />);
  expect(component).toMatchSnapshot();
});
