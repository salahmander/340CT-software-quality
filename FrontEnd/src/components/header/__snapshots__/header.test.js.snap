// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly in "debug" mode and match the snapshot 1`] = `
<div
  className="header"
>
  <div
    className="logo"
  >
    <button
      className="homeButton"
      onClick={[Function]}
    >
      <img
        alt="Acer"
        src="Acer.png"
      />
    </button>
  </div>
  <div
    className="header-center"
  >
    <button
      className="laptopButton"
      onClick={[Function]}
    >
       Laptop 
    </button>
    <button
      className="desktopButton"
      onClick={[Function]}
    >
       Desktop 
    </button>
    <span
      onClick={[Function]}
    >
      <button
        className="loginButton"
        type="Logout"
      >
         Login 
      </button>
    </span>
    <span
      onClick={[Function]}
    />
  </div>
</div>
`;
